kairon/api/app/routers/history.py

```diff
@@ -158,75 +196,74 @@ async def engaged_users_trend(month: int = Query(default=6, ge=1, le=6),
-async def new_users_trend(month: int = Query(default=6, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def new_users_trend(from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+                          to_date: datetime.date = datetime.date.today(),
+                          current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the counts of new users of the bot for previous months
     """
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/trends/users/new',
-        {'month': month}
+        {'from_date': from_date.strftime("%d-%m-%Y"), 'to_date': to_date.strftime("%d-%m-%Y")}
     )
 
 
 @router.get("/metrics/trend/conversation/success", response_model=Response)
-async def complete_conversation_trend(month: int = Query(default=6, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def complete_conversation_trend(from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+                                      to_date: datetime.date = datetime.date.today(),
+                                      current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the counts of successful conversations of the bot for previous months
     """
     fallback_action, nlu_fallback_action = DataUtility.load_fallback_actions(current_user.get_bot())
+    print(fallback_action, nlu_fallback_action)
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/trends/conversations/success',
-        {'month': month, 'action_fallback': fallback_action, 'nlu_fallback': nlu_fallback_action}
+        {'from_date': from_date.strftime("%d-%m-%Y"), 'to_date': to_date.strftime("%d-%m-%Y"),
+         'action_fallback': fallback_action, 'nlu_fallback': nlu_fallback_action}
     )
 
 
 @router.get("/metrics/trend/user/retention", response_model=Response)
-async def retention_trend(month: int = Query(default=6, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def retention_trend(from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+                          to_date: datetime.date = datetime.date.today(),
+                          current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the counts of user retention percentages of the bot for previous months
     """
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/trends/users/retention',
-        {'month': month}
```

Comment:
why this api has been removed?

Reply:
I haven't removed it. I just moved it to line no 54.

Reply:
why?